code/modules/vehicles/vehicle.dm

```diff
@@ -64,8 +66,8 @@
 
 
 //BUCKLE HOOKS
-/obj/vehicle/unbuckle_mob(force = 0)
-	if(buckled_mob)
+/obj/vehicle/unbuckle_mob(mob/living/buckled_mob,force = 0)
+	if(istype(buckled_mob))
```

Comment:
in that case why only check the typ in vehicle/unbuckle_mob() but not in atom/movable/proc/unbukle_mob() ?